File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    const map = {};
    
    for (let i=0; i < nums.length; i++) {
        
        let value = nums[i];
        let diff = target - value;
        
        if (diff in map) {
            return [map[diff], i]
        } else {
            map[value] = i;
        }
    }
    
};

File: 1-two-sum/1-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int n = nums[i];
            int diff = target - n;
            
            if(map.containsKey(n)) {
                return new int[]{map.get(n), i};
            } else {
                map.put(diff, i);
            }
        }
        return new int[0];
    }
}